### PR TITLE
feat(ocm): Add an option to decouple ocm config from k8s plugin

### LINKED ISSUE #48: Feature: OCM plugin: decouple backend auth from kuberentes plugin
**Is your feature request related to a problem? Please describe.**
Kuberentes plugin dependency may not be desired for multiple reasons:

- Adding a cluster to kubernetes plugin expects workload discovery on the cluster. If this is not permitted by a role, Backstage raises errors. Discovering workloads on the Hub cluster may not be desired.
- Accessing the Hub requires ServiceAccount access - it's accessed by the backend server at all times, therefore no identity provider can be used. That results in no RBAC applied, all users see the same/all data from the Hub cluster via Kuberentes plugin.

**Describe the solution you'd like**
Provide a different connection method. Consider changing the config schema to a partial of [`kubernetes-backend` cluster type](https://github.com/backstage/backstage/blob/38c306dc7f3ca9cf707dd768ac3c9b7fb5de6fd3/plugins/kubernetes-backend/config.d.ts#L37-L61):

```yaml
# app-config.yaml
ocm:
  # either use kubernetes
  name: # required if url not defined, string, current setup - match a cluster from kuberenetes plugin config
  # if name is not defined, use direct setup
  url: # required if name not defined, string, hub cluster api endpoint
  skipTLSVerify: # optional, bool, defaults to false
  skipMetricsLookup: # optional, bool, defaults to false
  caData: # optional, str, defaults to null
  caFile: # optional, str, defaults to null
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by SamoKopecky**
Resolves #48 

Change the `ocm-backend` config schema to allow for an explicit configuration of the ocm plugin without a requirement for the Kubernetes plugin configuration.

Add `serviceAccountToken` field to the hub configuration scheme since we don't have any other way to access the hub. We can then remove this in the future when we have proper RBAC.

Update/create tests for additional functions that are created in order to parse the configuration file.

Options like `skipMetricsLookup` and `caFile` are omitted since we don't even support these options in the code.

Update: Update documentation for standalone OCM configuration

### CODE REVIEW

**Comment by SamoKopecky** [comment]
/hold
Blocked since documentation needs to be udpated once #44 is merged.

File: plugins/ocm-backend/src/helpers/config.ts
Comment (tumido) [inline plugins/ocm-backend/src/helpers/config.ts:39]: Why would we do this here? We should call this function only if we already know the user chose `ocm.cluster` way. What's the reason for this call? Like.. is an attempt to raise error on invalid config?

File: plugins/ocm-backend/src/helpers/config.ts
Comment (tumido) [inline plugins/ocm-backend/src/helpers/config.ts:54]: Where if ever the hub cluster name is used?

**Review by tumido — CHANGES_REQUESTED** [review]

File: plugins/ocm-backend/schema.d.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/schema.d.ts:20]: We need it because it is used in `router.ts` to normalize the cluster name if the queried cluster is the hub.

File: plugins/ocm-backend/schema.d.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/schema.d.ts:6]: That is correct, but I think we should still leave it there. Some other frontend plugin might use this backend plugin and might need this configuration in the future.

File: plugins/ocm-backend/src/helpers/kubernetes.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/src/helpers/kubernetes.ts:4]: The old name was better, reverting the change.

File: plugins/ocm-backend/src/helpers/config.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/src/helpers/config.ts:35]: Yea thats a much better idea, I created a function that decides what config path to use in the latest commit.

File: plugins/ocm-backend/src/helpers/config.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/src/helpers/config.ts]: Changed in the latest commit.

File: plugins/ocm-backend/src/helpers/config.ts
Comment (SamoKopecky) [reply plugins/ocm-backend/src/helpers/config.ts]: This was stupid on my part, fixed in the latest commit by getting the name straight from the `ocm.cluster` config.

File: plugins/ocm/README.md
Comment (tumido) [inline plugins/ocm/README.md:84]: As per the schema this should be:

```
ocm:
  cluster:
    name: <name>
```

I think what you have it the docs here, is actually easier to understand. WDYT? Should the schema change or the docs?

**Review by tumido — CHANGES_REQUESTED** [review]
1 last change remaining, otherwise it LGTM.

File: plugins/ocm/README.md
Comment (SamoKopecky) [reply plugins/ocm/README.md:84]: I agree it makes is easier to understand. I changed it in the schema but also had to make some code changes. See the last commit.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
